### PR TITLE
Fix #434 support for radiasoft/sirepo#6636

### LINKED ISSUE #434: support for sirepo#6636
FileNotFoundError needs to be matched by pkio.exception_is_not_found.

### PR DESCRIPTION
**Pull request by robnagler**
- pkdexc accepts exc_info arg
- Add FileNotFound to exception_is_not_found
- add pkinspect.caller_func_name
- added pkunit.unbound_localhost_tcp_port and LOCALHOST_IP

### CODE REVIEW

**Comment by e-carlin** [comment]
Can you fix the title? I don't think #434 is the correct issue.

File: pykern/pkunit.py [hunk 1]
```diff
@@ -278,6 +283,29 @@ def file_eq(expect_path, *args, **kwargs):
     _FileEq(expect_path, *args, **kwargs)
 
 
+def unbound_localhost_tcp_port(start, stop):
+    """Looks for AF_INET SOCK_STREAM port for which bind fails
```
Comment (e-carlin) [inline pykern/pkunit.py:287]: Shouldn't it be "...for which bind __succeeds__" or maybe better "...port that we are able to bind to"?

Comment (robnagler) [reply pykern/pkunit.py:287]: If the bind fails, the port is "unbound" and therefore free to use as a server.

File: pykern/pkunit.py [hunk 2]
```diff
@@ -278,6 +283,29 @@ def file_eq(expect_path, *args, **kwargs):
...
+        with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
+            s.bind((LOCALHOST_IP, int(port)))
+        return port
+
+    for p in random.sample(range(start, stop), 10):
+        try:
+            return _check_port(p)
+        except Exception:
+            pass
+    raise ValueError(f"unable find port in range={start}-{stop} ip={LOCALHOST_IP}")
```
[the remark this answers is not in the document]
Comment (robnagler) [reply pykern/pkunit.py]: ee50221369

File: pykern/pkunit.py (hunk 1 above)
Comment (e-carlin) [reply pykern/pkunit.py:287]: I guess I don't understand what "bind fails" means? Fails implies some sort of error to me. From the code below it looks to me that if something is already bound to the ip:port then bind "fails". 

```bash
~$ lsof -i -P | grep 12345
~$ python <<EOF
> import socket

with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
    s.bind(("127.0.0.1", 12345))
    print("after bind")
EOF

> > > > > after bind
~$ python -m http.server 12345 &
[1] 26826
~$ Serving HTTP on 0.0.0.0 port 12345 (http://0.0.0.0:12345/) ...

~$ lsof -i -P | grep 12345
python  26826 vagrant    3u  IPv4 1547964      0t0  TCP *:12345 (LISTEN)
~$ python <<EOF
> import socket

with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
    s.bind(("127.0.0.1", 12345))
    print("after bind")
EOF

> > > > > Traceback (most recent call last):
  File "<stdin>", line 4, in <module>
OSError: [Errno 98] Address already in use
```

Comment (robnagler) [reply pykern/pkunit.py]: b3b10bb646. boolean confusion, sorry

**Review by e-carlin — APPROVED** [review]

**Pull request merged**